src/shared/libmount-util.c

```diff
@@ -38,3 +38,23 @@ int libmount_parse(
         *ret_iter = TAKE_PTR(iter);
         return 0;
 }
+
+int libmount_is_leaf(
+                struct libmnt_table *table,
+                struct libmnt_fs *fs) {
+        int r;
+
+        _cleanup_(mnt_free_iterp) struct libmnt_iter *iter_children = NULL;
+        iter_children = mnt_new_iter(MNT_ITER_FORWARD);
+
+        if (!iter_children)
+                return log_oom();
+
+        /* We care only whether it exists, it is unused */
+        _unused_ struct libmnt_fs *child;
+        r = mnt_table_next_child_fs(table, iter_children, fs, &child);
```

Comment:
i wished mnt_table_next_child_fs() whould just allow us to pass NULL as last argument here, but i guess it does not. :-(

Reply:
Looking at the source, it does de-reference it even if NULL.

Reply:
→ https://github.com/util-linux/util-linux/issues/2039

Reply:
our wishes were heard by karel. But I guess we need to retain compat with old libmount versions for a longer time, so nothing to change on our side